### DISCUSSION TITLE
Register handlers manually?

### DISCUSSION THREAD

**Discussion by SebastianStehle** [LuckyPennySoftware/MediatR#799] [Q&A]
Hi,

I am actually migrating a project to mediatr, because I would like to leverage the pipeline.

My project has an existing command structure already, but implemented in services: https://github.com/notifo-io/notifo/blob/main/backend/src/Notifo.Domain/Apps/IAppStore.cs#L29

I thought I can just implement the handler interface instead: https://github.com/notifo-io/notifo/blob/mediatr/backend/src/Notifo.Domain/Apps/AppStore.cs#L115

But I have no idea how to register this type. When I register as `IRequestHandler<AppCommand, App?>` interface it does not get discovered.

I guess I could add a separate command handler, but this would actually just create more code and I do not see the advantage yet. I would have to keep the upsert method in my IAppStore and just have an wrapper over it.

I user the normal assembly scanning process I would have duplicate instances.

**Comment by jbogard** [comment]
MediatR does not care about DI registration. It only resolves using a container. You do whatever you need to do to register types (there are samples in this repo).
